model-server/src/main/kotlin/org/modelix/model/server/handlers/ModelReplicationServer.kt

```diff
@@ -153,6 +154,7 @@ class ModelReplicationServer(
         val repositoryId = try {
             RepositoryId(repository)
         } catch (e: IllegalArgumentException) {
+            // TODO Olekz How does the "invalid-request" should look?
```

Comment:
```
 * @param details the detailed message to expose to the caller
 * @param typeSuffix A detailed type making this bad request instance uniquely identifiable. [PROBLEM_NAMESPACE] will
 *                   automatically be prepended. This field should be written in kebab-case.
 * @param cause The causing exception for the bad request or null if none.
 */
class BadRequestException(details: String, typeSuffix: String, cause: Throwable? = null) : HttpException(
    HttpStatusCode.BadRequest,
    title = "Bad request",
    details = details,
    type = problemType(typeSuffix),
    cause = cause,
)
```

@languitar  What should be the `typeSuffix` when creating a `BadRequestException`.
Should each occurrence have a different `typeSuffix`?

Reply:
`typeSuffix` is meant to be a unique identifier for the type of problem resulting in this error condition. For instance, something like "out-of-credit". Cf. https://www.rfc-editor.org/rfc/rfc9457.html#name-type

Reply:
Ok, I created subclasses of `HttpException` instead of using `BadRequestException`.


(Creating all these subclasses instead of having functions that create instances of `HttpException` is so Java-like 😆)

Reply:
We can still decide which way we want to go. I guess consistent with other frameworks would be to at least have classes mapping to HTTP response codes.

Reply:
I'm fine classes.